### PR TITLE
Play vinyl's tracks in sequence 1️⃣➡2️⃣➡3️⃣

### PR DESCRIPTION
**Pull request by dmcrodrigues**

### CODE REVIEW

**Comment by RuiAAPeres** [comment]
The `func test_Vinyl_multiple_uniquely()` modifications are wrong. We are now calling `expectation.fullFill()` will potentially be called twice:

```
let mockedTrackedNotFound = MockedTrackedNotFoundErrorHandler { expectation.fulfill() }
```

And here:

```
            if numberOfCalls == 2 {
 +                expectation.fulfill()
 +            }
```

**Comment by RuiAAPeres** [comment]
This:

```
-        turnatable.dataTaskWithRequest(request2, completionHandler: checker).resume()
 -        turnatable.dataTaskWithRequest(request2, completionHandler: checker).resume()
 -        turnatable.dataTaskWithRequest(request2, completionHandler: checker).resume()
```

Was on purpose as well. So you were the same request wouldn't called.

**Comment by RuiAAPeres** [comment]
Looking at the `test_Vinyl_withTrackOrder()`, how does the test prove that the tracks were chosen in the correct order? If I change the implementation to just pick the first track, I think it will pass as well.

File: VinylTests/TurntableTests.swift [hunk 1]
```diff
@@ -210,4 +210,45 @@ class TurntableTests: XCTestCase {
...
+            guard let httpResponse = response as? NSHTTPURLResponse else {
+                fatalError("response should be a `NSHTTPURLResponse`")
+            }
+            
+            XCTAssertTrue(httpResponse.statusCode == 200)
+            XCTAssertTrue(taskData!.isEqualToData(data))
+            XCTAssertTrue(httpResponse.allHeaderFields as! HTTPHeaders == headers)
+            XCTAssertTrue(httpResponse.URL?.absoluteString == "http://feelGoodINC.com")
+            XCTAssertTrue(httpResponse.URL?.absoluteString == "http://feelGoodINC.com")
+        }
```
Comment (RuiAAPeres) [inline VinylTests/TurntableTests.swift]: Is this intended?

```
+            XCTAssertTrue(httpResponse.URL?.absoluteString == "http://feelGoodINC.com")
+            XCTAssertTrue(httpResponse.URL?.absoluteString == "http://feelGoodINC.com")
```

**Comment by dmcrodrigues** [comment]
Oh boy 🙄

**Comment by RuiAAPeres** [comment]
Sorry for the delay, was messing around with the code, didn't had a full mental picture of what was going on. Looks good! 🐰

**Pull request merged**